stock_picking_invoice_product_group/models/account_invoice.py

```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Alex Comba - Agile Business Group
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, fields
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    related_picking_ids = fields.Many2many(
```

Comment:
Well, having this field serves for all purposes, so duplication doesn't make change. Even more, the module is broken when you merge some invoices (with the corresponding module), because you can merge several invoices from different pickings, so I consider the change is needed (although it requires migration scripts like your post-init-hook and adapt dependent modules).